Source/LocalNetAppChat/LocalNetAppChat.Domain/Shared/ReceivedMessage.cs

```diff
@@ -1,3 +1,4 @@
 ﻿namespace LocalNetAppChat.Domain.Shared;
 
-public record ReceivedMessage(long Id, DateTime Timestamp, LnacMessage Message);
\ No newline at end of file
+public record ReceivedMessage(long Id, DateTime Timestamp, LnacMessage Message);
+public record ReceivedDirectMessage(long Id, DateTime Timestamp, LnacMessage Message, string Receiver);
```

Comment:
Please do not do that. Instead extend the normal ReceivedMessage with a Receiver. The Receiver should be empty, when it is not a direct message and filled when it is. This way the handling code should be a lot less.